### PR TITLE
Add Clinical Tracker CA Secrets

### PR DESCRIPTION
**Pull request by kris-szlapa**
## Summary

- Routine Change

### Details

This pull request implements the integration of Clinical Tracker CA Secrets

### CODE REVIEW

File: privateCA/create_certs.sh [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+    --query 'Stacks[0].Outputs[?OutputKey==`'"${SECRET_OUTPUT_PREFIX}"'ClientCertSecret`].OutputValue' --output text)
+CLIENT_SANDBOX_KEY_ARN=$(aws cloudformation describe-stacks \
+    --stack-name account-resources \
+    --query 'Stacks[0].Outputs[?OutputKey==`'"${SECRET_OUTPUT_PREFIX}"'ClientSandboxKeySecret`].OutputValue' --output text)
+CLIENT_SANDBOX_CERT_ARN=$(aws cloudformation describe-stacks \
+    --stack-name account-resources \
+    --query 'Stacks[0].Outputs[?OutputKey==`'"${SECRET_OUTPUT_PREFIX}"'ClientSandboxCertSecret`].OutputValue' --output text)
+TRUSTSTORE_BUCKET_ARN=$(aws cloudformation describe-stacks \
+    --stack-name account-resources \
+    --query 'Stacks[0].Outputs[?OutputKey==`'"${SECRET_OUTPUT_PREFIX}"'TrustStoreBucket`].OutputValue' --output text)
```
Comment (anthony-nhs) [inline privateCA/create_certs.sh:186]: you dont need SECRET_OUTPUT_PREFIX here - we have one bucket that all CA certificates will go to

File: cloudformation/account_resources.yml [hunk 2]
```diff
@@ -839,6 +893,40 @@ Outputs:
...
+    Export:
+      Name: !Join [":", [!Ref "AWS::StackName", "ClinicalTrackerCACertSecret"]]
+
+  ##################################################
+  # Clinical Tracker Client Secrets Outputs
+  ##################################################
+  ClinicalTrackerClientKeySecret:
+    Description: ARN of the clinical tracker client key secret
+    Value: !GetAtt ClinicalTrackerClientKeySecret.Id
+
```
Comment (anthony-nhs) [inline cloudformation/account_resources.yml:917]: can you add an export for this as you have done for ClinicalTrackerCACertSecret

File: cloudformation/account_resources.yml [hunk 3]
```diff
@@ -839,6 +893,40 @@ Outputs:
...
+  ClinicalTrackerClientCertSecret:
+    Description: ARN of the clinical tracker client cert secret
+    Value: !GetAtt ClinicalTrackerClientCertSecret.Id
```
Comment (anthony-nhs) [inline cloudformation/account_resources.yml:920]: can you add an export for this as you have done for ClinicalTrackerCACertSecret

File: cloudformation/account_resources.yml [hunk 4]
```diff
@@ -839,6 +893,40 @@ Outputs:
...
+
+  ClinicalTrackerClientSandboxKeySecret:
+    Description: ARN of the clinical tracker client key secret for sandbox
+    Value: !GetAtt ClinicalTrackerClientSandboxKeySecret.Id
```
Comment (anthony-nhs) [inline cloudformation/account_resources.yml:924]: can you add an export for this as you have done for ClinicalTrackerCACertSecret

File: cloudformation/account_resources.yml [hunk 5]
```diff
@@ -839,6 +893,40 @@ Outputs:
...
+
+  ClinicalTrackerClientSandboxCertSecret:
+    Description: ARN of the clinical tracker client cert secret for sandbox
+    Value: !GetAtt ClinicalTrackerClientSandboxCertSecret.Id
```
Comment (anthony-nhs) [inline cloudformation/account_resources.yml:928]: can you add an export for this as you have done for ClinicalTrackerCACertSecret

File: privateCA/create_certs.sh [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+        --secret-string file://${KEYS_DIR}/apigee_client_cert_sandbox.key
+    aws secretsmanager put-secret-value \
+        --secret-id ${CLIENT_SANDBOX_CERT_ARN} \
+        --secret-string file://${CERTS_DIR}/apigee_client_cert_sandbox.pem
+
+    echo "Going to create new truststore files on S3"
+    read -p "Press any key to resume or press ctrl+c to exit ..."
+
+    aws s3 cp ${CERTS_DIR}/truststore.pem s3://${TRUSTSTORE_BUCKET_NAME}/truststore.pem
+    aws s3 cp ${CERTS_DIR}/sandbox_truststore.pem s3://${TRUSTSTORE_BUCKET_NAME}/sandbox-truststore.pem
```
Comment (anthony-nhs) [inline privateCA/create_certs.sh:269]: these two lines need changing as for prescription-tracker it should upload to s3://${TRUSTSTORE_BUCKET_NAME}/clinical-tracker-truststore.pem and clinical-tracker-sandbox-truststore.pem

File: cloudformation/account_resources.yml (hunk 2 above)
Comment (kris-szlapa) [reply cloudformation/account_resources.yml:920]: The export section has been added.

File: cloudformation/account_resources.yml (hunk 3 above)
Comment (kris-szlapa) [reply cloudformation/account_resources.yml:923]: The export section has been added.

File: cloudformation/account_resources.yml (hunk 4 above)
Comment (kris-szlapa) [reply cloudformation/account_resources.yml:930]: The export section has been added.

File: cloudformation/account_resources.yml (hunk 5 above)
Comment (kris-szlapa) [reply cloudformation/account_resources.yml:940]: The export section has been added.

File: privateCA/create_certs.sh (hunk 1 above)
Comment (kris-szlapa) [reply privateCA/create_certs.sh]: The SECRET_OUTPUT_PREFIX has been removed.

File: privateCA/create_certs.sh (hunk 6 above)
Comment (kris-szlapa) [reply privateCA/create_certs.sh]: The script has been changed as follows
```
cat ${BACKUP_CERTS_DIR}/s3_${FILE_PREFIX}_truststore.pem ${CERTS_DIR}/${CA_NAME}.pem > ${CERTS_DIR}/${FILE_PREFIX}_truststore.pem
cat ${BACKUP_CERTS_DIR}/s3_${FILE_PREFIX}_sandbox_truststore.pem ${CERTS_DIR}/${CA_NAME}.pem > ${CERTS_DIR}/${FILE_PREFIX}_sandbox_truststore.pem
```
Where
```
$ ./create_certs.sh -e dev -a "clinical tracker"
Going to create mutual TLS certs with these details
AWS_PROFILE: 
CERT_PREFIX: dev-
SECRET_OUTPUT_PREFIX: ClinicalTracker
BUCKET_PREFIX: clinical-tracker
FILE_PREFIX: clinical_tracker
DRY_RUN: true
Press any key to resume or press ctrl+c to exit ...
```

**Review by anthony-nhs — APPROVED** [review]

**Review by MatthewPopat-NHS — APPROVED** [review]

**Pull request merged**